parser/tavor.go

```diff
@@ -1565,102 +1564,33 @@ func (p *tavorParser) parseTypedTokenDefinition(variableScope *token.VariableSco
 		return zeroRune, err
 	}
 
-	var tok token.Token
-	usedArguments := make(map[string]struct{})
-
-	switch typ {
-	case "Int":
-		rawFrom, okFrom := arguments["from"]
-		rawTo, okTo := arguments["to"]
-
-		from := 0
-		if okFrom {
-			from, err = strconv.Atoi(rawFrom)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"from\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-		}
-
-		to := math.MaxInt32
-		if okTo {
-			to, err = strconv.Atoi(rawTo)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"to\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-		}
-
-		step := 1
-
-		if raw, ok := arguments["step"]; ok {
-			step, err = strconv.Atoi(raw)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"step\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-
-			usedArguments["step"] = struct{}{}
-		}
-
-		usedArguments["from"] = struct{}{}
-		usedArguments["to"] = struct{}{}
-
-		tok = primitives.NewRangeIntWithStep(from, to, step)
-	case "Sequence":
-		start := 1
-		step := 1
-
-		if raw, ok := arguments["start"]; ok {
-			start, err = strconv.Atoi(raw)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"start\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
-		}
-
-		if raw, ok := arguments["step"]; ok {
-			step, err = strconv.Atoi(raw)
-			if err != nil {
-				return zeroRune, &token.ParserError{
-					Message:  "\"step\" needs an integer value",
-					Type:     token.ParseErrorInvalidArgumentValue,
-					Position: p.scan.Pos(),
-				}
-			}
+	// find the constructor for the current type
+	createTok, found := token.GetTypedConstructor(typ)
```

Comment:
(Seems like the UI swallowed my comment...)
On seeing this I am wondering if the argument parser should be used here as a parameter, and instead of returning a function which returns a token, just return a token then. The call to the function with a argument parser has to be done anyway so why not just put it into the typed token code. What do you think?